### ISSUE TITLE
About memory missing location information

### ISSUE THREAD

**Issue by LzhinFdu** [Beomi/InfiniTransformer#23]
I noticed that the memory retrieval and update happens before 'apply_rotary_pos_emb'. Wondering whether the memory lacking location information would confuse the model's perception of the order of historical information?

**Comment by Lazy3valuation** [comment]
From the readme: "Can train 'infinite' context -- check train.gemma.infini.noclm.1Mseq.sh with 1x H100 80G (with AdamW optimizer, No gradient checkpointing)". However I can train it with 12GB with 8b quantization and a segment size of 400.

**Comment by LzhinFdu** [comment]
I can also run through training. However, the current training results are not very good. I'm trying to train further

**Comment by pengshuang** [comment]
I also have same question, can you solve it ?

**Comment by LzhinFdu** [comment]
You can try to adjust the memory retrieval process to the end of 'apply_rotary_pos_emb' and compare the training performance. However, I did not try it further.

**Comment by pengshuang** [comment]
Thanks for your response.

**Comment by lihua8848** [comment]
Can this retain location information?
